tests/datavolume_test.go

```diff
@@ -134,6 +136,254 @@ var _ = Describe("DataVolume Integration", func() {
+				dataVolume, err := virtClient.CdiClient().CdiV1alpha1().DataVolumes(namespace).Get(name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				Expect(dataVolume.OwnerReferences).ToNot(BeEmpty())
+
+				if dataVolume.Status.Phase != cdiv1.Succeeded {
+					return fmt.Errorf("Waiting on DataVolume to enter succeeded phase")
+				}
+				return nil
+			}, 160*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		}
+
+		dataVolumeIsSuccessAndNotOwned := func(name, namespace string) {
+			Eventually(func() error {
+				dataVolume, err := virtClient.CdiClient().CdiV1alpha1().DataVolumes(namespace).Get(name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				Expect(dataVolume.OwnerReferences).To(BeEmpty())
+
+				if dataVolume.Status.Phase != cdiv1.Succeeded {
+					return fmt.Errorf("Waiting on DataVolume to enter succeeded phase")
+				}
+				return nil
+			}, 160*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		}
+
+		pvcExists := func(name, namespace string) {
+			Eventually(func() error {
+				_, err := virtClient.CoreV1().PersistentVolumeClaims(namespace).Get(name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				return nil
+			}, 160*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		}
+
+		waitForDeletionVM := func(name, namespace string) {
+			Eventually(func() bool {
+				_, err := virtClient.VirtualMachine(namespace).Get(name, &metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					return true
+				}
+				return false
+			}, 100*time.Second, 1*time.Second).Should(BeTrue())
+		}
+
+		waitForDeletionVMI := func(name, namespace string) {
+			Eventually(func() bool {
+				_, err := virtClient.VirtualMachineInstance(namespace).Get(name, &metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					return true
+				}
+				return false
+			}, 100*time.Second, 1*time.Second).Should(BeTrue())
+		}
+
+		waitForDeletionDataVolume := func(name, namespace string) {
+			Eventually(func() bool {
+				_, err := virtClient.CdiClient().CdiV1alpha1().DataVolumes(namespace).Get(name, metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					return true
+				}
+				return false
+			}, 100*time.Second, 1*time.Second).Should(BeTrue())
+		}
+
+		waitForDeletionPVC := func(name, namespace string) {
+			Eventually(func() bool {
+				_, err = virtClient.CoreV1().PersistentVolumeClaims(namespace).Get(name, metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					return true
+				}
+				return false
+			}, 100*time.Second, 1*time.Second).Should(BeTrue())
+
+		}
+
+		AfterEach(func() {
+			if vmJson != "" {
+				err = os.Remove(vmJson)
+				Expect(err).ToNot(HaveOccurred())
+				vmJson = ""
+			}
+
+			deleteIfExistsVM(vm.Name, vm.Namespace)
+			deleteIfExistsVMI(vm.Name, vm.Namespace)
+			deleteIfExistsDataVolume(dataVolumeName, vm.Namespace)
+		})
+
+		It("deleting VM with cascade=true should automatically delete DataVolumes and VMI owned by VM.", func() {
```

Comment:
test_id:

It("[test_id:837]deleting VM with cascade=true should automatically delete DataVolumes and VMI owned by VM.", func() {

Reply:
fixed